Vic2ToHoI4/Source/HOI4World/Names/CultureNames.h

```diff
@@ -0,0 +1,54 @@
+#ifndef CULTURE_NAMES_H
+#define CULTURE_NAMES_H
+
+
+
+#include "Parser.h"
+#include <string>
+#include <vector>
+
+
+
+namespace HoI4
+{
+
+class CultureNames: commonItems::parser
+{
+  public:
+	explicit CultureNames(std::istream& theStream);
+
+	void convertNamesToUTF8();
+
+	[[nodiscard]] auto takeMaleNames() { return std::move(maleNames); }
```

Comment:
this is black magic to me. You're feeding a move of something to the caller, so what remains after you move it?
I usually give const refs which are cheap and don't actually copy anything, and don't really understand this.

Reply:
You've understood correctly. And it is black magic, they're called zombie objects.

But in this case it's what I want. The CultureNames class is a throwaway importer, so I'm taking the names from it and letting the zombie get cleaned up when the class falls out of scope.